src/plugins/foxPage/hooks/useDefiOpportunity.ts

```diff
@@ -0,0 +1,46 @@
+import { DefiProvider } from 'features/defi/contexts/DefiManagerProvider/DefiCommon'
+import { EarnOpportunityType } from 'features/defi/helpers/normalizeOpportunity'
+import { useMemo, useState } from 'react'
+import { useFoxEthLpBalances } from 'pages/Defi/hooks/useFoxEthLpBalances'
+import { useFoxFarmingBalances } from 'pages/Defi/hooks/useFoxFarmingBalances'
+import { selectFeatureFlags } from 'state/slices/selectors'
+import { useAppSelector } from 'state/store'
+
+import { ExternalOpportunity } from '../FoxCommon'
+
+export const useDefiOpportunity = (opportunity: ExternalOpportunity) => {
+  const [defiOpportunity, setDefiOpportunity] = useState<EarnOpportunityType | null>(null)
+  const { opportunity: foxEthLpOpportunity } = useFoxEthLpBalances()
+  const { opportunities: foxFarmingOpportunities } = useFoxFarmingBalances()
+  const featureFlags = useAppSelector(selectFeatureFlags)
+
+  useMemo(() => {
```

Comment:
`useMemo` is the wrong construct here, we're not assigning this to anything and have a memoized expression that's unused. 
Either remove the `defiOpportunity` state field and return from this, or make this a `useEffect`